src/query/expression/src/types/string.rs

```diff
@@ -156,6 +158,47 @@ impl ValueType for StringType {
     fn column_memory_size(col: &Self::Column) -> usize {
         col.data().len() + col.offsets().len() * 8
     }
+
+    fn cmp(op: &SelectOp) -> fn(Self::ScalarRef<'_>, Self::ScalarRef<'_>) -> bool {
```

Comment:
We just need to keep :

```
fn cmp(Self::ScalarRef<'_>, Self::ScalarRef<'_>) -> Ordering {..}
```
The latter functions could be removed, see `register_simple_domain_type_cmp`